app/components/mfa_selection_component.js

```diff
@@ -0,0 +1,17 @@
+function clearPhoneSelectionError() {
+  const error = document.getElementById('phone_error');
+  const invalid = document.querySelector('label.checkbox__invalid');
+  if (error) {
+    error.style.display = 'none';
+  }
+  if (invalid) {
+    invalid.classList.remove('checkbox__invalid');
+  }
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+  const checkboxes = document.getElementsByName('two_factor_options_form[selection][]');
```

Comment:
This is pretty specific to the one form, and not really a reusable component. Also, `ValidatedField` is already implementing quite a bit around field validation that seems to be repeated here.

Could we target a class or element name instead? Also, if we apply validity to the input using `input.setCustomValidity`, the default behavior of `ValidatedField` might be able to automatically handle parts of showing / hiding error messages (see `PhoneInputComponent` for example), as well as do some client-side validation, so we don't have to do a full server form submission to validate the input.

```ts
const checkboxes = document.querySelectorAll('.some-checkbox-class');
const phoneCheckbox = checkboxes.find((checkbox) => checkbox.name === 'phone');
checkboxes.forEach((checkbox) => {
  checkbox.addEventListener('input', () => {
    const checked = checkboxes
      .filter((checkbox) => checkbox.checked)
      .map((checkbox) => checkbox.name);
    const hasPhone = checked.includes('phone');
    const hasOther = hasPhone && checked.length > 1;
    phoneCheckbox.setCustomValidity(hasPhone && !hasOther ? 'some error' : '');
  });
});
```

Reply:
So I couldn't reuse validated field for the reasons I mentioned in my other comment above.

I'm targeting a pretty specific label name to get a list of the labels I want to target already. Not sure what using the querySelectorAll would add.

One note is that the intended behavior is for the error to clear when the user selects another input (or unselects phone input), but not come back if they unselect the other input (or reselect just the phone input). It would only show again if they tried to resubmit with just phone selected.

Reply:
I think the key thing for me here is that a ViewComponent implies some amount of reusability, or at least leverages the separation of the class from the view for managing a complex view model. I don't really see either of those here, so I'd almost wonder if a partial makes more sense, located somewhere alongside other MFA form views.

Calling `document.getElementsByName('two_factor_options_form[selection][]')` will only work for [this one specific form](https://github.com/18F/identity-idp/blob/8a5f85e93d7ccbb5667ce26257e78efd7c67fe2d/app/views/two_factor_authentication/options/index.html.erb#L12), so the idea with `querySelectorAll` was to at least make it generic enough if that form name were to change, or if the component/partial were to be used anywhere else.

Reply:
Sounds good. I turned it into a partial since I think this component and validation requirement is super specific to this form and I move the js into a javascript/pack.